pkg/controllers/capr/managesystemagent/managesystemagent.go

```diff
@@ -371,7 +369,116 @@ func installer(cluster *rancherv1.Cluster, secretName string) []runtime.Object {
 		})
 	}
 
-	return append([]runtime.Object{plan, windowsPlan}, objs...)
+	return append(plans, objs...)
+}
+
+func winsUpgradePLan(cluster *rancherv1.Cluster, env []corev1.EnvVar, secretName string) *upgradev1.Plan {
+	winsUpgradeImage := strings.SplitN(settings.WinsAgentUpgradeImage.Get(), ":", 2)
+	winsVersion := "latest"
+	if len(winsUpgradeImage) == 2 {
+		winsVersion = winsUpgradeImage[1]
+	}
+
+	return &upgradev1.Plan{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Plan",
+			APIVersion: upgradeAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "system-agent-upgrader-windows",
+			Namespace: namespaces.System,
+			Annotations: map[string]string{
+				upgradeDigestAnnotation: "spec.upgrade.envs,spec.upgrade.envFrom",
+			},
+		},
+		Spec: upgradev1.PlanSpec{
+			Concurrency: 10,
+			Version:     winsVersion,
+			Tolerations: []corev1.Toleration{
+				{
+					Operator: corev1.TolerationOpExists,
+				},
+			},
+			NodeSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      corev1.LabelOSStable,
+						Operator: metav1.LabelSelectorOpIn,
+						Values: []string{
+							"windows",
+						},
+					},
+				},
+			},
+			ServiceAccountName: systemAgentUpgraderServiceAccountName,
+			Upgrade: &upgradev1.ContainerSpec{
+				Image: image.ResolveWithCluster(winsUpgradeImage[0], cluster),
+				Env:   env,
+				SecurityContext: &corev1.SecurityContext{
+					WindowsOptions: &corev1.WindowsSecurityContextOptions{
+						HostProcess:   toBoolPointer(true),
+						RunAsUserName: toStringPointer("NT AUTHORITY\\SYSTEM"),
+					},
+				},
+				EnvFrom: []corev1.EnvFromSource{{
+					SecretRef: &corev1.SecretEnvSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secretName,
+						},
+					},
+				}},
+			},
+		},
+	}
+}
+
+func toBoolPointer(x bool) *bool {
+	return &x
+}
+
+func toStringPointer(x string) *string {
+	return &x
+}
+
+// currentVersionResolvesGH5551 determines if the given rke2 version
+// has fixed the RKE2 bug outlined in GH-5551. Windows SUC plans cannot be delivered
+// to clusters running versions containing this bug. This function can be removed
+// when v1.31.x is the lowest supported version offered by Rancher.
+func currentVersionResolvesGH5551(version string) bool {
+
+	// remove leading v and trailing distro identifier
+	v := strings.TrimPrefix(version, "v")
+	verSplit := strings.Split(v, "+")
+	if len(verSplit) != 2 {
+		return false
+	}
+
+	curSemVer, err := semver.NewVersion(verSplit[0])
+	if err != nil {
+		return false
+	}
+
+	// GH-5551 was resolved in 1.30.4, before the release of
+	// 1.31, so we know that all versions higher than 1.30 will
+	// not contain this bug.
+	if curSemVer.Minor() >= 31 {
+		return true
+	}
+
+	switch curSemVer.Minor() {
+	case 30:
+		return curSemVer.GreaterThan(GH5551FixedVersions[0]) || curSemVer.Equal(GH5551FixedVersions[0])
```

Comment:
```suggestion
		return curSemVer.GreaterThanEqual(GH5551FixedVersions[0])
```
https://pkg.go.dev/github.com/Masterminds/semver/v3#Version.GreaterThanEqual

Reply:
Ah cool, looks like that was introduced in 3.3.0 and we're still using 3.2.1. I've updated the dependency and the function used